pkg/api/api.go

```diff
@@ -401,9 +401,9 @@ func (hs *HTTPServer) registerRoutes() {
 		}
 
 		apiRoute.Group("/plugins", func(pluginRoute routing.RouteRegister) {
-			pluginRoute.Get("/:pluginId/dashboards/", reqOrgAdmin, routing.Wrap(hs.GetPluginDashboards))
+			pluginRoute.Get("/:pluginId/dashboards/", reqOrgAdmin, hs.CheckEnabled, routing.Wrap(hs.GetPluginDashboards))
 			pluginRoute.Post("/:pluginId/settings", authorize(ac.EvalPermission(pluginaccesscontrol.ActionWrite, pluginIDScope)), routing.Wrap(hs.UpdatePluginSetting))
-			pluginRoute.Get("/:pluginId/metrics", reqOrgAdmin, routing.Wrap(hs.CollectPluginMetrics))
+			pluginRoute.Get("/:pluginId/metrics", reqOrgAdmin, hs.CheckEnabled, routing.Wrap(hs.CollectPluginMetrics))
```

Comment:
Not sure that we need to protect this one

Reply:
my logic was why a plugin return metrics if it's disabled but maybe I am missing something. Should I remove it?

Reply:
I think the only thing that cares about this is Prometheus (note we as Grafana are not even currently using this actively) - so if someone out there is scraping for plugin metrics, some scrapes will now fail because of this. So I think it's probably fine to leave it out in case it does cause an issue for someone. Probably not a big deal though tbh